superset/columns/models.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Column model.
+
+This model was introduced in SIP-68 (https://github.com/apache/superset/issues/14909),
+and represents a "column" in a table or dataset. In addition to a column, new models for
+tables, metrics, and datasets were also introduced.
+
+These models are not fully implemented, and shouldn't be used yet.
+"""
+
+import sqlalchemy as sa
+from flask_appbuilder import Model
+
+from superset.models.helpers import (
+    AuditMixinNullable,
+    ExtraJSONMixin,
+    ImportExportMixin,
+)
+
+
+class Column(
+    Model, AuditMixinNullable, ExtraJSONMixin, ImportExportMixin,
+):
+    """
+    A "column".
+
+    The definition of column here is overloaded: it can represent a physical column in a
+    database relation (table or view); a computed/derived column on a dataset; or an
+    aggregation expression representing a metric.
+    """
+
+    __tablename__ = "sl_columns"
```

Comment:
Semantic layer. I wanted to use `columns` and `tables`, but those are taken by `TableColumn` and `SqlaTable`. Originally I had the migration rename these tables, but it would force us to have downtime when upgrading, so I opted for using a prefix instead.

I'm open to other suggestions for a prefix here. :)